### DISCUSSION TITLE
Clarification about use of janitor

### DISCUSSION THREAD

**Discussion by mig5** [ory/hydra#3107] [Q&A] [answered]
Hi,

I've been reading https://www.ory.sh/docs/hydra/cli/hydra-janitor and it's not clear to me if it's safe to run on a production instance where people have saved 'remember me' on login, saved consent etc.

If I run the command `janitor[...] --tokens --requests --grants` will it remove any active 'remembered' sessions and saved consents? Or is it just for things that are no longer needed by any active users? Note: in my case I am not using any refresh tokens, e.g for 'offline' scope use..

If I were to use the time-based parameters e.g `--keep-if-younger xx` would this mean 'only users who have saved their consent for an oAuth.2.0 client in the last xx days will have their consent saved' but older ones won't?

Basically I'm not clear on the ramifications if I were to run the command. I don't want to run the command and then get complaints that the users have to accept the Consent options for each oAuth2.0 client again next time they log in. 

If it's just about deleting old login/consent request challenge codes as part of those flows (which are never used again after submitting the login/consent forms) then that's cool :) 

I note that the page says "will cleanup any *expired* oauth2 tokens as well as login/consent requests." (my emphasis on 'expired'). So does that mean it *only* ever works on things it considers to have already expired, regardless of what other time control flags are sent to the janitor command? If so that gives me confidence it's safe to use :)

Thanks!

**Comment by vinckr** [comment, marked as answer]
> So does that mean it only ever works on things it considers to have already expired, regardless of what other time control flags are sent to the janitor command?

AFAIK it will only remove expired OAuth2 tokens regardless of flags. so even when you run `--keep-if-younger xx` it should only remove the expired OAuth2 tokens.

Can you please confirm this @Benehiko 🙏

**Reply by Benehiko** [reply]
@vinckr Yes :) 

@mig5 
It only deletes expired requests / tokens. The `--keep-if-younger` flag is there for auditing purposes. Sometimes it is useful to hold data from the past 30 days, but not for the past 2 years ;)

Please also note that this command could take a long time to run if you have millions of records. I would suggest running this when the load on your database is the lowest and **also always make a backup** before running janitor.

**Reply by mig5** [reply]
Thanks @Benehiko !
